scripts/metadata_table.py

```diff
@@ -96,475 +125,1137 @@
+                                       module=module, var_dict=var_dict,
+                                       known_ddts=known_ddts, logger=logger)
+                self.__sections.append(sect)
+            # end if
+        else:
+            self.__table_name = None
+            self.__table_type = None
+            if dependencies is not None:
+                perr = "dependencies not allowed as argument when reading file"
+                raise ParseInternalError(perr)
+            # end if
+            if relative_path:
+                perr = "relative_path not allowed as argument when reading file"
+                raise ParseInternalError(perr)
+            # end if
+            if var_dict is not None: # i.e., not even an empty dict
+                perr = "var_dict not allowed as argument when reading file"
+                raise ParseInternalError(perr)
+            if module:
+                perr = "module not allowed as argument when reading file"
+                raise ParseInternalError(perr)
+            # end if
+            if known_ddts is None:
+                known_ddts = list()
+            # end if
+            self.__start_context = ParseContext(context=self.__pobj)
+            self.__init_from_file(known_ddts, logger)
+        # end if
+
+    def __init_from_file(self, known_ddts, logger):
+        """ Read the table preamble, assume the caller already figured out
+        the first line of the header using the header_start method."""
+        curr_line, _ = self.__pobj.next_line()
+        in_properties_header = True
+        skip_rest_of_section = False
+        self.__dependencies = list() # Default is no dependencies
+        # Process lines until the end of the file or start of the next table.
+        while ((curr_line is not None) and
+               (not MetadataTable.table_start(curr_line))):
+            if MetadataSection.header_start(curr_line):
+                in_properties_header = False
+            # end if
+            if blank_metadata_line(curr_line):
+                curr_line, _ = self.__pobj.next_line()
+            elif in_properties_header:
+                # Process the properties in this table header line
+                for prop in _parse_config_line(curr_line, self.__pobj):
+                    # Manually parse name, type, and table properties
+                    key = prop[0].strip().lower()
+                    value = prop[1].strip()
+                    if key == 'name':
+                        self.__table_name = value
+                    elif key == 'type':
+                        if value not in TABLE_TYPES:
+                            self.__pobj.add_syntax_err("metadata table type",
+                                                       token=value)
+                        # end if
+                        self.__table_type = value
+                    elif key == 'dependencies':
+                        if value.lower() == "none" or value.strip() == '':
```

Comment:
Why is a blank `dependencies` line an error here?

Reply:
capgen's metadata parser was creating a dependencies list that contained an item `[ ' ' ]` - this produced an error.

Reply:
That will not be happening once you merge #395 (and should not have been happening with the change you made).